spec/lib/locasms/client_spec.rb

```diff
@@ -2,218 +2,165 @@
+        allow(rest_client).to receive(:get).and_return({})
+
+        client.deliver 'given message', '11988889991', '11988889992', '11988889993', url_callback: 'something'
+
+        expect(rest_client).to have_received(:get).once.with(:sendsms, some_callback_args)
       end
+    end
 
-      it 'uses default callback' do
-        client = LocaSMS::Client.new :login, :password, rest_client: rest_client, url_callback: 'default'
+    it 'uses default callback' do
+      client = described_class.new :login, :password, rest_client: rest_client, url_callback: 'default'
 
-        expect(client).to receive(:numbers)
-          .once
-          .with(%i[a b c])
-          .and_return('XXX')
+      allow(rest_client).to receive(:get).and_return({})
 
-        expect(rest_client).to receive(:get)
-          .once
-          .with(:sendsms, msg: 'given message', numbers: 'XXX', url_callback: 'default')
-          .and_return({})
+      client.deliver 'given message', '11988889991', '11988889992', '11988889993'
 
-        client.deliver 'given message', :a, :b, :c
-      end
+      expect(rest_client).to have_received(:get).once.with(:sendsms, default_callback_args)
     end
   end
 
   describe '#deliver_at' do
-    it 'Should send SMS' do
-      expect(subject).to receive(:numbers)
-        .once
-        .with(%i[a b c])
-        .and_return('XXX')
-
-      expect(LocaSMS::Helpers::DateTimeHelper).to receive(:split)
-        .once
-        .with(:datetime)
-        .and_return(%w[date time])
-
-      expect(rest_client).to receive(:get)
-        .once
-        .with(:sendsms, msg: 'given message', numbers: 'XXX', jobdate: 'date', jobtime: 'time', url_callback: nil)
-        .and_return({})
-
-      subject.deliver_at 'given message', :datetime, :a, :b, :c
+    let(:base_args) do
+      {
+        msg: 'given message',
+        numbers: '11988889991,11988889992,11988889993',
+        jobdate: '10/10/2020',
+        jobtime: '10:10'
+      }
     end
 
-    it 'Should not send SMS' do
-      expect(subject).to receive(:numbers)
-        .once
-        .with(%i[a b c])
-        .and_raise(LocaSMS::Exception)
+    it 'sends SMS' do
+      allow(rest_client).to receive(:get).and_return({})
 
-      expect(LocaSMS::Helpers::DateTimeHelper).to receive(:split)
-        .once
-        .with(:datetime)
-        .and_return(%w[date time])
+      client.deliver_at 'given message', '2020-10-10 10:10', '11988889991', '11988889992', '11988889993'
 
-      expect(rest_client).not_to receive(:get)
+      expect(rest_client).to have_received(:get).once.with(:sendsms, base_args.merge(url_callback: nil))
+    end
 
-      expect { subject.deliver_at('given message', :datetime, :a, :b, :c) }.to raise_error(LocaSMS::Exception)
+    context 'when receive an error' do
+      let(:wrong_deliver_at) { -> { client.deliver_at('given message', '2020-10-10 10:10', '1', '2', '3') } }
+
+      it { expect(wrong_deliver_at).to raise_error(LocaSMS::Exception) }
```

Comment:
Eu não sei.. Eu preferia a versão anterior com a chamada direto dentro do it.. But, it's your call

Reply:
essa versão anterior?

```ruby
expect { subject.deliver_at('given message', :datetime, :a, :b, :c) }.to raise_error(LocaSMS::Exception)
```